cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -289,6 +289,13 @@ public synchronized void updateProperties(Id.Namespace namespaceId, NamespaceMet
       }
     }
 
+    if (config != null && config.getHiveDatabase() != null) {
+      // if a root directory was given for update and it's not same as existing one throw exception
+      if (!config.getHiveDatabase().equals(existingMeta.getConfig().getHiveDatabase())) {
+        throw new BadRequestException(String.format("%s cannot be updated.", NamespaceConfig.HIVE_DATABASE));
```

Comment:
Can you include what it can't be updated from and to? "Updates to %s are not allowed. Cannot update from %s to %s."